File: 단계별로 풀어보기/정수론 및 조합론/5086/배수와 약수.py
import sys

def init():
    a, b = map(int,sys.stdin.readline().split())
    while a != 0 and b != 0:
        if a%b == 0:
            sys.stdout.write("multiple\n")
        elif b%a == 0:
            sys.stdout.write("factor\n")
        else:
            sys.stdout.write("neither\n")
        a, b = map(int,sys.stdin.readline().split())
init()

File: 단계별로 풀어보기/정수론 및 조합론/5086/배수와 약수.cpp
#include <iostream>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	
	int a, b;
	while (cin >> a >> b) {
		if (a == 0 && b == 0)
			break;
        if(a%b == 0){
            cout << "multiple\n";
        } else if(b%a == 0){
            cout << "factor\n";
        } else{
            cout << "neither\n";
        }
	}
	return 0;
}
